UnitsNet/UnitDefinitions/AmountOfSubstance.json

```diff
@@ -2,6 +2,9 @@
   "Name": "AmountOfSubstance",
   "BaseUnit": "Mole",
   "XmlDoc": "Mole is the amount of substance containing Avagadro's Number (6.02 x 10 ^ 23) of real particles such as molecules,atoms, ions or radicals.",
+  "BaseDimensions": {
+    "N": 1
```

Comment:
I think N can be removed altogether, since we store Value with the quantity?
So `2 Feet = 2 x 0.3048 Meter` where Value = 2 and M = 0.3048. Have I got it right?

Just a thought for a future PR, but it might be cool to add dimensions for all units too - not just the base units. This way I think we can possibly generate conversions between units instead of specifying from/to conversions functions in JSON. I have no idea how logarithmic and other non-linear units will work with such conversions, but it's probably doable.

Reply:
I am a bit confused by your comment. N is the [SI base dimension symbol](https://en.wikipedia.org/wiki/SI_base_unit) for amount of substance. It is the base unit so only one base dimension is filled with a value of 1.

I have added the base dimensions for all units. For example Molarity is the amount of a substance in a volume: N / L³. It is represented in Molarity.json as such:

```json
  "BaseDimensions": {
    "L": -3,
    "N": 1
  },
```

And you can write this code in my branch to prove the division works correctly:

``` csharp
var result = AmountOfSubstance.BaseDimensions / Volume.BaseDimensions;
Assert.True( Molarity.BaseDimensions == result );
```
As I noted in my request: these are the dimensions that we can generate the conversions from. However knowing the base dimensions is very useful even without that work being done yet. However this also takes us a step closer to being able to do so!

Reply:
You are absolutely right, I was confusing "Amount of substance" with "Amount" or "Value".

Regarding 
> add dimensions for all units

If I understand this right, `BaseDimensions` only describes the quantity's base unit - not all other units. So, for `Length`, base dimensions read `M=1` for `Length`'s base unit `Meter`. This allows us to generate operator overloads such as `Area (M=2) = Length (M=1) * Length (M=1)`. However, it does not allow us to generate the conversion from `Meter` to `Foot`, without also defining the dimensions of the `Foot` unit, right?

Reply:
Correct it's only the base dimensions that the derived units boil down to.

Length for example is defined in base units as follows:

```json
  "BaseDimensions": {
    "L": 1
  },
```
I think you're confusing the base dimensions with the unit. M is the [Base Dimension Symbol](https://en.wikipedia.org/wiki/SI_base_unit) for Mass, not meters. 

Area is defined in base dimensions as Length * Length or L^2 (L: 2). So Length (L: 1) when multiplied adds the base dimensions giving you L: 2. You'd then look this up in a map of base dimensions -> quantity.  L:2 being area.

The conversion from Meter to Feet would still be in the Length quantity. Nothing else is necessary. Base dimensions apply to ALL units in length (feet, meters, fathoms etc.) They all have a base dimension of Length = 1 (L: 1).

Reply:
Thanks, I was indeed confused by this. As you say, I was of the understanding that the dimensions could be used to express *all units* - not just the base unit of each quantity. The confusion arise from that it is always the *base unit* of the quantity that expresses the dimensions, so for `Area` the base unit is `m²` and thus `L=2`.  In my mind, we could then somehow represent feet and inches the same way, since most units is a simple linear transformation of the base unit, but maybe that gets into the same territory why chose to have two conversion functions for `from` and `to`, since this function is not always trivial to invert.

Anyway, I think I'm onboard with the correct thinking now :-)

Reply:
The word base in base dimensions is different than base as in base units. All derived units (like area) are derived from some combination of the 7 base dimensions. Length is a base dimension. Feet or meters are not. They are units that represent the length base dimension.

The base dimensions *are* expressed for **all** units inside a quantity, not just the base unit.

Area, whether m² or in² or ft², all have base dimensions of L=2.